model/admin/db.go

```diff
@@ -0,0 +1,98 @@
+package admin
+
+import (
+	"github.com/evergreen-ci/evergreen/db"
+	"github.com/evergreen-ci/evergreen/db/bsonutil"
+	"github.com/pkg/errors"
+	"gopkg.in/mgo.v2/bson"
+)
+
+const (
+	Collection       = "admin"
+	systemSettingsID = "0"
+)
+
+var (
+	IdKey     = bsonutil.MustHaveTag(AdminSettings{}, "Id")
+	BannerKey = bsonutil.MustHaveTag(AdminSettings{}, "Banner")
+
+	ServiceFlagsKey  = bsonutil.MustHaveTag(AdminSettings{}, "ServiceFlags")
+	TaskDispatchKey  = bsonutil.MustHaveTag(ServiceFlags{}, "TaskDispatchDisabled")
+	HostinitKey      = bsonutil.MustHaveTag(ServiceFlags{}, "HostinitDisabled")
+	MonitorKey       = bsonutil.MustHaveTag(ServiceFlags{}, "MonitorDisabled")
+	NotificationsKey = bsonutil.MustHaveTag(ServiceFlags{}, "NotificationsDisabled")
+	AlertsKey        = bsonutil.MustHaveTag(ServiceFlags{}, "AlertsDisabled")
+	TaskrunnerKey    = bsonutil.MustHaveTag(ServiceFlags{}, "TaskrunnerDisabled")
+	RepotrackerKey   = bsonutil.MustHaveTag(ServiceFlags{}, "RepotrackerDisabled")
+	SchedulerKey     = bsonutil.MustHaveTag(ServiceFlags{}, "SchedulerDisabled")
+)
+
+var settingsQuery = db.Query(bson.M{IdKey: systemSettingsID})
+
+func GetSettingsFromDB() (*AdminSettings, error) {
+	settings := &AdminSettings{}
+	query := db.Q{}
+	query = query.Filter(settingsQuery)
+	err := db.FindOneQ(Collection, query, settings)
+	if err != nil {
+		// if the settings document doesn't exist, return the defaults
```

Comment:
I considered this, but it felt cleaner to me to return the defaults. The document will get inserted the first time someone changes one of the settings